test/preview/components/builders/test_answers_builders.py

```diff
@@ -0,0 +1,156 @@
+        data = component.to_dict()
+        assert data == {
+            "type": "AnswersBuilder",
+            "init_parameters": {"pattern": "pattern", "reference_pattern": "reference_pattern"},
+        }
+
+    @pytest.mark.unit
+    def test_from_dict(self):
+        data = {
+            "type": "AnswersBuilder",
+            "init_parameters": {"pattern": "pattern", "reference_pattern": "reference_pattern"},
+        }
+        component = AnswersBuilder.from_dict(data)
+        assert component.pattern == "pattern"
+        assert component.reference_pattern == "reference_pattern"
+
+    @pytest.mark.unit
+    def test_run_unmatching_input_len(self):
+        component = AnswersBuilder()
+        with pytest.raises(ValueError):
+            component.run(queries=["query"], replies=[["reply1"], ["reply2"]], metadata=[[]])
+
+    def test_run_without_pattern(self):
+        component = AnswersBuilder()
+        answers = component.run(queries=["test query"], replies=[["Answer: AnswerString"]], metadata=[[{}]])
+        assert len(answers) == 1
+        assert len(answers[0]) == 1
+        assert answers[0][0].data == "Answer: AnswerString"
+        assert answers[0][0].metadata == {}
+        assert answers[0][0].query == "test query"
+        assert answers[0][0].documents == []
+        assert isinstance(answers[0][0], GeneratedAnswer)
+
+    def test_run_with_pattern_with_capturing_group(self):
+        component = AnswersBuilder(pattern=r"Answer: (.*)")
+        answers = component.run(queries=["test query"], replies=[["Answer: AnswerString"]], metadata=[[{}]])
+        assert len(answers) == 1
+        assert len(answers[0]) == 1
+        assert answers[0][0].data == "AnswerString"
+        assert answers[0][0].metadata == {}
+        assert answers[0][0].query == "test query"
+        assert answers[0][0].documents == []
+        assert isinstance(answers[0][0], GeneratedAnswer)
+
+    def test_run_with_pattern_without_capturing_group(self):
+        component = AnswersBuilder(pattern=r"'.*'")
+        answers = component.run(queries=["test query"], replies=[["Answer: 'AnswerString'"]], metadata=[[{}]])
+        assert len(answers) == 1
+        assert len(answers[0]) == 1
+        assert answers[0][0].data == "'AnswerString'"
+        assert answers[0][0].metadata == {}
+        assert answers[0][0].query == "test query"
+        assert answers[0][0].documents == []
+        assert isinstance(answers[0][0], GeneratedAnswer)
+
+    def test_run_with_pattern_with_more_than_one_capturing_group(self):
+        component = AnswersBuilder(pattern=r"Answer: (.*), (.*)")
+        with pytest.raises(ValueError, match="contains multiple capture groups"):
+            component.run(queries=["test query"], replies=[["Answer: AnswerString, AnswerString2"]], metadata=[[{}]])
+
+    def test_run_with_pattern_set_at_runtime(self):
+        component = AnswersBuilder()
```

Comment:
Nitpick: let's put another pattern here in the init to make sure `AnswerBuilder` uses the right one when both are set.